src/keystore.js

```diff
@@ -8,15 +8,25 @@ const LRU = require('lru')
 const Buffer = require('safe-buffer/').Buffer
 const { verifier } = require('./verifiers')
 
+function createStore (path = './keystore') {
+  if (fs && fs.mkdirSync) {
+    fs.mkdirSync(path, { recursive: true })
+  }
+  return level(path)
+}
+
 class Keystore {
-  constructor (store) {
-    if (!store || typeof store === 'string') {
-      const path = store || './keystore'
-      if (fs && fs.mkdirSync) fs.mkdirSync(path, { recursive: true })
-      store = level(path)
-    }
-    this._store = store
-    this._cache = new LRU(100)
+  constructor (input = {}) {
+    if (typeof input === 'string') {
+      this._store = createStore(input)
+    } else if (typeof input.open === 'function') {
```

Comment:
Can you explain the purpose of checking `input.open`? Are you treating the options object as the store? Imo we should expect the store to be passed as `options.store`

Reply:
Hi @shamb0t,

From my understanding, the current API allows three cases:
1. no arguments
2. a string which must be a path
3. a `level` store

So yes the first `if` block checks if `input` is a `level` store (case .3).
And the second one checks if `input` is a string, i.e. a path (case .2).

The intend is to avoid breaking changes.

However, if breaking changes are OK on your side I can review the constructor implementation accordingly.

Best regards

Reply:
Makes sense :+1: